### PR TITLE
Added custom filter for satellites

### LINKED ISSUE #2060: SATELLITES Plugin
Can a filter be added to just show satellites brighter than a specified magnitude?

I realize that in order to work for all sats, they all have to have a Standard Magnitude. Perhaps for any object where Standard Mag is unknown, just assign a default of 4 if RCS is Large, 6 if Medium, 10 if small.

Thank you.

### PR DESCRIPTION
**Pull request by alex-w**
### Description
In this pull request I added ability use customisation filter the list of satellites by few criterias, e.g. apogee, perigee, inclination, etc. The discussion about requirements for this filter was via email. The UI and behaviour is similar to Satellites filter in HeavenSat program.

Fixes #2060 (issue)

### Screenshots:
stellarium-006

Satellites filter in HeavenSat:
unnamed

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the [code style](http://stellarium.org/doc/head/codingStyle.html) of this project.
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (header file)
- [ ] I have updated the respective chapter in the Stellarium User Guide
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by alex-w** [comment]
OK, the PR is ready to review (need SUG update)

**Comment by gzotti** [comment]
Just checking...

File: plugins/Satellites/src/Satellites.hpp [hunk 1]
```diff
@@ -572,6 +615,101 @@ public slots:
 	//! Save the current satellite catalog to disk.
 	void saveCatalog(QString path=QString());
 
+	//! Set whether custom filter 'known standard magnitude' enabled.
+	//! Emits customFilterChanged() if the value changes.
+	void setFlagCFKnownStdMagnitude(bool b);
+	bool getFlagCFKnownStdMagnitude() { return Satellite::flagCFKnownStdMagnitude; }
+
+	//! Set whether custom filter 'apogee' enabled.
+	//! Emits customFilterChanged() if the value changes.
```
Comment (gzotti) [inline plugins/Satellites/src/Satellites.hpp:624]: Actually it emits the signal in any case. A super-clean setter should test the flag and only set+emit if it changes. So, either this (preferred), or the documentation must drop "if the value changes" (multiple times).

**Review by gzotti — APPROVED** [review]
The setters should be perfectioned as stated to fit the documentation. Else it seems to work.

**Comment by JAY-Githb** [comment]
Hi. 
I can't find that update. 
Can you furnish a direct link to the Windows version? 
Thank you. 

Jay Respler 
Advanced Business Machines Co. 
609-662-7438 
***@***.*** 
Monroe Township, New Jersey

**Comment by alex-w** [comment]
> Hi. I can't find that update. Can you furnish a direct link to the Windows version? 

Binary packages are not prepared yet

**Comment by JAY-Githb** [comment]
I saw the proposed Satellite Filter. 
It looks very nice. 

Jay Respler 
Advanced Business Machines Co. 
609-662-7438 
***@***.*** 
Monroe Township, New Jersey

**Comment by JAY-Githb** [comment]
As I noted elsewhere, GREAT JOB!

Jay

**Pull request merged**